presto-main/src/test/java/com/facebook/presto/operator/TestExchangeClient.java

```diff
@@ -320,6 +321,80 @@ public void testClose()
         assertEquals(clientStatus.getHttpRequestState(), "not scheduled", "httpRequestState");
     }
 
+    @Test
+    public void testRemoveRemoteSource()
+            throws Exception
+    {
+        DataSize maxResponseSize = new DataSize(1, Unit.BYTE);
+        MockExchangeRequestProcessor processor = new MockExchangeRequestProcessor(maxResponseSize);
+
+        URI location1 = URI.create("http://localhost:8081/foo");
+        TaskId taskId1 = TaskId.valueOf("foo");
+        URI location2 = URI.create("http://localhost:8082/bar");
+        TaskId taskId2 = TaskId.valueOf("bar");
+
+        processor.addPage(location1, createPage(1));
+        processor.addPage(location1, createPage(2));
+        processor.addPage(location1, createPage(3));
+
+        ExchangeClient exchangeClient = new ExchangeClient(
+                new DataSize(1, Unit.BYTE),
+                maxResponseSize,
+                1,
+                new Duration(1, TimeUnit.MINUTES),
+                true,
+                new TestingHttpClient(processor, newCachedThreadPool(daemonThreadsNamed("test-%s"))),
```

Comment:
We create several thread pools in test cases in this class, we can create one and share across tests I think given that we had issues in the past with tests related to memory usage. If you do that please do it in a separate commit.

Reply:
Yeah, let's create a test wide, and manage it with `@BeforeClass` and `@AfterClass`